pkg/webui/components/key-value-map/entry.js

```diff
@@ -0,0 +1,90 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import { Col, Row } from 'react-grid-system'
+import bind from 'autobind-decorator'
+
+import PropTypes from '../../lib/prop-types'
+import Input from '../input'
+import Button from '../button'
+
+@bind
+class Entry extends React.Component {
+
+  handleRemoveButtonClicked (event) {
+    this.props.onRemoveButtonClick(this.props.index, event)
+  }
+
+  handleKeyChanged (newKey) {
+    this.props.onChange(this.props.index, { key: newKey })
+  }
+
+  handleValueChanged (newValue) {
+    this.props.onChange(this.props.index, { value: newValue })
+  }
+
+  render () {
+    return (
+      <Row key={this.props.index}>
```

Comment:
I think using the grid container is not appropriate here. You sould rather use a container with `display: flex` and `flex-direction: column` and then size the individual elements using percentage values for `width`.

Reply:
If you want I can make a PR for this branch to take care of this.